### ISSUE TITLE
no-unsafe-finally doesn't catch nested violations

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#327]
```
try {
  /* */
} finally {
  try {
    /* */
  } finally {
    throw "oh no";
  }
}
```

From https://rdambrosio016.github.io/rust/2020/09/18/pure-ast-based-linting-sucks.html
